config/routes.rb

```diff
@@ -184,41 +184,45 @@
   end
 
   namespace :schools do
-    resource :dashboard, controller: :dashboard, only: :show, path: "/"
-    resource :choose_programme, controller: :choose_programme, only: %i[show create], path: "choose-programme" do
-      get :advisory
-
-      get :confirm_programme, path: "confirm-programme"
-      get :choice_saved_design_our_own, path: "design-your-programme"
-      get :choice_saved_no_early_career_teachers, path: "no-early-career-teachers"
-      post :save_programme, path: "save-programme"
-      get :success
-    end
-
-    resources :cohorts, only: :show do
-      resources :partnerships, only: :index
-      resource :programme, only: %i[edit], controller: "choose_programme"
+    resources :dashboard, controller: :dashboard, only: :show, path: "/", param: :school_id
 
-      resources :participants, only: %i[index show] do
-        get :edit_details, path: "edit-details"
-        get :edit_mentor, path: "edit-mentor"
-        put :update_mentor, path: "update-mentor"
+    scope "/:school_id" do
+      resource :choose_programme, controller: :choose_programme, only: %i[show create], path: "choose-programme" do
+        get :advisory
 
-        collection do
-          multistep_form :add, Schools::AddParticipantForm, controller: :add_participants
-        end
+        get :confirm_programme, path: "confirm-programme"
+        get :choice_saved_design_our_own, path: "design-your-programme"
+        get :choice_saved_no_early_career_teachers, path: "no-early-career-teachers"
+        post :save_programme, path: "save-programme"
+        get :success
       end
 
-      namespace :core_programme, path: "core-programme" do
-        resource :materials, only: %i[edit update show] do
-          get :info
-          get :success
-        end
-      end
+      resources :cohorts, only: :show, param: :cohort_id do
+        member do
+          resources :partnerships, only: :index
+          resource :programme, only: %i[edit], controller: "choose_programme"
+
+          resources :participants, only: %i[index show] do
+            get :edit_details, path: "edit-details"
+            get :edit_mentor, path: "edit-mentor"
+            put :update_mentor, path: "update-mentor"
+
+            collection do
+              multistep_form :add,
+                             Schools::AddParticipantForm, controller: :add_participants
```

Comment:
If we do multiline, maybe we want brackets around it?

Reply:
I'm happy to inline this again, snuck in in a merge